Foundation/Network/IPv6.hs

```diff
@@ -155,3 +135,122 @@ toTuple (IPv6 hi low) =
     w6 = low .>>. 32
     w7 = low .>>. 16
     w8 = low
+
+-- | IPv6 Parser as described in RFC4291
+--
+-- for more details: https://tools.ietf.org/html/rfc4291.html#section-2.2
+--
+-- which is exactly:
+--
+-- ```
+--     ipv6ParserPreferred
+-- <|> ipv6ParserIPv4Embedded
+-- <|> ipv6ParserCompressed
+-- ```
+--
+ipv6Parser :: (Sequential input, Element input ~ Char)
+           => Parser input IPv6
+ipv6Parser =  ipv6ParserPreferred
```

Comment:
I'm not sure it's a good idea to use `Alternative` here and instead not relying on (much slower) backtracking (restarting parsing at the beginning everytime). It's fine right now though.